### PR TITLE
Support Dida365（Ticktick China）

### PR DESCRIPTION
**Pull request by MrSnake0208**
1. Use Dida365=True or False
2. Another login method for Dida365

### CODE REVIEW

**Comment by MrSnake0208** [comment]
The reason why changing ticktick.com to dida365.com directly does not work for the TickTick China version is that dida365 requires headers and there are authentication issues. It is necessary to use base64 encryption for client id and secret. After consulting with official personnel, I made adjustments to the code:
1. Added dida365 options for users to decide whether or not to use dida365.
2. Modified the way of obtaining access token from dida365.
As a beginner, the formatting may be a bit messy, so you can format it uniformly after testing.

File: ticktick/oauth2.py [hunk 1]
```diff
@@ -244,7 +269,8 @@ def _post(self, url, **kwargs):
             RunTimeError: If the request could not be completed.
         """
 
-        response = self.session.post(url, **kwargs)
+        response = self.session.request("POST",url, **kwargs)
+        print(response.content)
```
Comment (C0rn3j) [inline ticktick/oauth2.py:273]: I think you left an accidental print statement in the commit here.

**Comment by chenyucheng0503** [comment]
hello, I noticed that you delete the branch that support Dida365. Is there any backup for it? thx!

**Comment by MrSnake0208** [comment]
> hello, I noticed that you delete the branch that support Dida365. Is there any backup for it? thx!

You can click the code above, that's what i edited

File: ticktick/oauth2.py (hunk 1 above)
Comment (MrSnake0208) [reply ticktick/oauth2.py:273]: Thx

**Pull request closed without merging**